packages/codecs/src/anchor.ts

```diff
@@ -1,120 +1,69 @@
+import { uint8ArrayAsBase64 } from './binary.js'
 
-/**
- * Describes all anchor statuses
- */
-export enum AnchorStatus {
-  NOT_REQUESTED = 0,
-  PENDING = 1,
-  PROCESSING = 2,
-  ANCHORED = 3,
-  FAILED = 4,
-  REPLACED = 5,
+export enum RequestStatusName {
+  PENDING = 'PENDING',
+  PROCESSING = 'PROCESSING',
+  COMPLETED = 'COMPLETED',
+  FAILED = 'FAILED',
+  READY = 'READY',
+  REPLACED = 'REPLACED',
 }
 
-export const AnchorProof = sparse(
-  {
-    chainId: string,
-    txHash: cidAsString,
-    root: cidAsString,
-    txType: optional(string),
-  },
-  'AnchorProof'
-)
-export type AnchorProof = TypeOf<typeof AnchorProof>
-
-export const AnchorCommit = strict(
-  {
-    id: cidAsString,
-    prev: cidAsString,
-    proof: cidAsString,
-    path: string,
-  },
-  'AnchorCommit'
-)
-export type AnchorCommit = TypeOf<typeof AnchorCommit>
-
-export const AnchorServicePending = strict(
+export const CommitPresentation = sparse(
   {
-    status: literal(AnchorStatus.PENDING),
-    streamId: streamIdAsString,
-    cid: cidAsString,
-    message: string,
+    content: optional(
+      sparse({ path: optional(string), prev: string, proof: optional(string) }, 'content')
+    ),
+    cid: string.pipe(cidAsString),
   },
-  'AnchorServicePending'
+  'CommitPresentation'
 )
-export type AnchorServicePending = TypeOf<typeof AnchorServicePending>
+export type CommitPresentation = TypeOf<typeof CommitPresentation>
 
-export const AnchorServiceProcessing = strict(
-  {
-    status: literal(AnchorStatus.PROCESSING),
-    streamId: streamIdAsString,
-    cid: cidAsString,
-    message: string,
-  },
-  'AnchorServiceProcessing'
-)
-export type AnchorServiceProcessing = TypeOf<typeof AnchorServiceProcessing>
+export const NotCompleteStatusName = union([
+  literal(RequestStatusName.PENDING),
+  literal(RequestStatusName.PROCESSING),
+  literal(RequestStatusName.FAILED),
+  literal(RequestStatusName.READY),
+  literal(RequestStatusName.REPLACED),
+])
+export type NotCompleteStatusName = TypeOf<typeof NotCompleteStatusName>
 
-export const AnchorServiceAnchored = sparse(
+export const NotCompleteCASResponse = sparse(
   {
-    status: literal(AnchorStatus.ANCHORED),
+    status: NotCompleteStatusName,
     streamId: streamIdAsString,
     cid: cidAsString,
     message: string,
-    anchorCommit: cidAsString,
-    witnessCar: optional(carAsUint8Array),
   },
-  'AnchorServiceAnchored'
+  'NotCompleteCASResponse'
 )
-export type AnchorServiceAnchored = TypeOf<typeof AnchorServiceAnchored>
+export type NotCompleteCASResponse = TypeOf<typeof NotCompleteCASResponse>
 
-export const AnchorServiceFailed = strict(
+export const CompleteCASResponse = type(
   {
-    status: literal(AnchorStatus.FAILED),
-    streamId: streamIdAsString,
-    cid: cidAsString,
-    message: string,
+    ...NotCompleteCASResponse.props,
+    status: literal(RequestStatusName.COMPLETED),
+    anchorCommit: CommitPresentation,
+    witnessCar: uint8ArrayAsBase64.pipe(carAsUint8Array),
```

Comment:
should witnessCar be optional, at least temporarily until all the CAS environments are returning it?  ie CDB-2519

Reply:
It is now.